pkgs/test_core/lib/src/util/os.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+import 'package:test_api/src/backend/operating_system.dart'; // ignore: implementation_imports
+
+/// Directories that are specific to OS X.
+///
+/// This is used to try to distinguish OS X and Linux in [currentOSGuess].
+final _macOSDirectories = {
```

Comment:
could be const?

Reply:
I wondered the same thing but I didn't look too closely at it. I don't know that we can say a priori that `const` would be an optimization, if it might not be read in some runs.

In any case I'm going to intentionally avoid implementation changes with these PRs shuffling code around.

Reply:
For me it's more about making sure collections like this aren't accidentally mutated. I get your point, though.